wait/exit.go

```diff
@@ -53,13 +53,16 @@ func ForExit() *ExitStrategy {
 	return NewExitStrategy()
 }
 
+func (ws *ExitStrategy) Timeout() *time.Duration {
+	return ws.timeout
+}
+
 // WaitUntilReady implements Strategy.WaitUntilReady
 func (ws *ExitStrategy) WaitUntilReady(ctx context.Context, target StrategyTarget) (err error) {
-	// limit context to exitTimeout
-	if ws.exitTimeout > 0 {
-		var cancelContext context.CancelFunc
-		ctx, cancelContext = context.WithTimeout(ctx, ws.exitTimeout)
-		defer cancelContext()
+	if ws.timeout != nil {
```

Comment:
What happens if the user sets timeout to zero? Reading the code of `context.WithTimeout`:

```golang
// WithTimeout returns WithDeadline(parent, time.Now().Add(timeout)).
//
// Canceling this context releases resources associated with it, so code should
// call cancel as soon as the operations running in this Context complete:
//
// 	func slowOperationWithTimeout(ctx context.Context) (Result, error) {
// 		ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
// 		defer cancel()  // releases resources if slowOperation completes before timeout elapses
// 		return slowOperation(ctx)
// 	}
func WithTimeout(parent Context, timeout time.Duration) (Context, CancelFunc) {
	return WithDeadline(parent, time.Now().Add(timeout))
}
```

It seems that the zero would be added to now, cancelling immediately. Is this assumption correct?

Reply:
Zero would only be used if explicitly defined by the user, which would be odd. The reason we use a pointer for `timeout` is to differentiate between an undefined (nil) state and user-defined (time.Duration) state.